kivy/uix/recyclegridlayout.py

```diff
@@ -220,22 +222,24 @@ def compute_visible_views(self, data, viewport):
         right = x + w
         top = y + h
         at_idx = self.get_view_index_at
-        # 'tl' is not actually 'top-left' unless 'orientation' is 'lr-tb'.
-        # But we can pretend it always is. Same for 'bl' and 'br'.
-        tl, __, bl, br = sorted((
+        tl, tr, bl, br = sorted((
             at_idx((x, y)),
             at_idx((right, y)),
             at_idx((x, top)),
             at_idx((right, top)),
         ))
 
         n = len(data)
+        if len({tl, tr, bl, br}) < 4:
+            # visible area is one row/column
+            return list(range(min(n, tl), min(n, br + 1)))
```

Comment:
This part doesn't have to create a list. I have to fix it.

Reply:
Hmm I suppose. The interface doesn't say that it needs to return a list so it should be fine.